src/bin/report_collector.rs

```diff
@@ -247,7 +275,7 @@ async fn ipa(
             ipa_query_config.per_user_credit_cap,
             ipa_query_config.attribution_window_seconds,
             ipa_query_config.max_breakdown_key,
-            &CappingOrder::CapOldestFirst,
+            &order,
```

Comment:
Can you just inline the logic on lines 258-262 here? I think that will be easier to read.